### PR TITLE
[SPARK-50353][SQL] Refactor ResolveSQLOnFile

### PR DESCRIPTION
**Pull request by mihailoale-db**
### What changes were proposed in this pull request?
Refactoring of `ResolveSQLOnFile` rule.

### Why are the changes needed?
Refactoring is needed for the Single-pass Analyzer project (please check [link](https://issues.apache.org/jira/browse/SPARK-49834))

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing tests.

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala [hunk 1]
```diff
@@ -47,6 +47,34 @@ import org.apache.spark.util.ArrayImplicits._
  * Replaces [[UnresolvedRelation]]s if the plan is for direct query on files.
  */
 class ResolveSQLOnFile(sparkSession: SparkSession) extends Rule[LogicalPlan] {
+  object UnresolvedRelationResolution {
```
Comment (dtenedor) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala:50]: I think you have to put this object outside of the class if you want any other code to be able to use it? Playing around with the Scala REPL, I am not able to refer to the nested object without creating an instance of the class.

```
Welcome to the Ammonite Repl 3.0.0-M2 (Scala 3.3.3 Java 17.0.12)
@ class c() {
    object d {
      val x = 42
    }
  } 
defined class c

@ val e = new c() 
e: c = ammonite.$sess.cmd0$c@7a48e6e2

@ e.d 
res2: _root_.ammonite.$sess.cmd1.e.d.type = ammonite.$sess.cmd0$c$d$@9d2723e

@ e.d.x 
res3: Int = 42

@ c.d.x 
-- [E008] Not Found Error: cmd4.sc:1:13 ----------------------------------------
1 |val res4 = c.d.x
  |           ^^^
  |value d is not a member of object ammonite.$sess.cmd0.c, but could be made available as an extension method.
  |
  |The following import might make progress towards fixing the problem:
  |
  |  import sourcecode.Text.generate
  |
Compilation Failed
```

**Review by dtenedor — APPROVED** [review]
LGTM after resolving the one comment

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala (hunk 1 above)
Comment (mihailoale-db) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala:50]: If we move it out of class we'll have to introduce some other changes e.g make `maybeSQLFile` public etc. I think it is better like this (we create an instance of class and use the object as a field of it). @dtenedor

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @mihailoale-db and @dtenedor for review.

**Pull request closed without merging**